source/OutfitterPanel.cpp

```diff
@@ -694,6 +697,10 @@ void OutfitterPanel::FailSell(bool toStorage) const
 	const string &verb = toStorage ? "uninstall" : "sell";
 	if(!planet || !selectedOutfit)
 		return;
+	else if(!selectedOutfit->CanSell() && !toStorage)
+		GetUI()->Push(new Dialog("You cannot " + verb + " this outfit."));
+	else if(!selectedOutfit->CanUninstall() && toStorage)
+		GetUI()->Push(new Dialog("You cannot " + verb + " this outfit."));
```

Comment:
If an outfit cannot be uninstalled, then selling the ship (the complete ship) should result in the Outfit not being added to the stock (the outfit gets sold with the ship, so it should not be available for separate purchase later).